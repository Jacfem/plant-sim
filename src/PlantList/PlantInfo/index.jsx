import React from 'react';
import { Link } from 'react-router-dom';

import './styles.css';

const PlantInfo = props => (
  <li key={props.plant.key}>
    <div className="plantContainer">
      <h3>{props.plant.name}:
        <Link to={`/plant/${props.plant.slug}`}>
          Try it!
        </Link>
      </h3>
      <p>Water:
        <span className="showOnHover">
          {props.plant.needs.water}
        </span>
      </p>
      <p>Days needed:
        <span className="showOnHover">
          {props.plant.needs.sunlight}
        </span>
      </p>
      <p>Fertilizer:
        <span className="showOnHover">
          {props.plant.needs.fertilizer}
        </span>
      </p>
    </div>
  </li>
);

// Add propTypes required here

export default PlantInfo;
