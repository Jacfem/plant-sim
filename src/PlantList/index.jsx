import React, { Component } from 'react';

import { seedData } from '../seedData';
import PlantInfo from './PlantInfo';

class PlantList extends Component {
  render() {
    return (
      <ul className="unstyledList">
        {seedData.map(plant =>
          (<PlantInfo plant={plant} key={plant.key} />))
        }
      </ul>
    );
  }
}

export default PlantList;
