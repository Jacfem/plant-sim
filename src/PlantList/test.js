import PlantList from './';

describe('<PlantList />', () => {
  function subject() {
    return Enzyme.shallow(<PlantList />); // takes no props, just seed data
  }

  describe('#render', () => {
    it('matches snapshot', () => {
      expect(subject()).toMatchSnapshot();
    });
  });
});
