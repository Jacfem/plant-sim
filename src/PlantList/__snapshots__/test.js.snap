// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantList /> #render matches snapshot 1`] = `
<ul
  className="unstyledList"
>
  <PlantInfo
    key="1"
    plant={
      Object {
        "key": "1",
        "name": "Herbs",
        "needs": Object {
          "fertilizer": 4,
          "sunlight": 5,
          "water": 6,
        },
        "picture": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR5BZubtgkC5zoKphMnbSWcOvH3dcRNnFutoN0nj6WF-07p0swM",
        "slug": "herbs",
      }
    }
  />
  <PlantInfo
    key="2"
    plant={
      Object {
        "key": "2",
        "name": "Carrots",
        "needs": Object {
          "fertilizer": 9,
          "sunlight": 8,
          "water": 10,
        },
        "picture": "https://hgtvhome.sndimg.com/content/dam/images/hgtv/fullset/2018/3/21/0/shutterstock_221550766_Lukas-Gojda_carrot-plants.jpg.rend.hgtvcom.616.411.suffix/1521655913719.jpeg",
        "slug": "carrots",
      }
    }
  />
  <PlantInfo
    key="3"
    plant={
      Object {
        "key": "3",
        "name": "Cucumbers",
        "needs": Object {
          "fertilizer": 7,
          "sunlight": 5,
          "water": 10,
        },
        "picture": "https://www.attainable-sustainable.net/wp-content/uploads/2018/02/growing-cucumbers.jpg",
        "slug": "cucumbers",
      }
    }
  />
</ul>
`;
